### PR TITLE
Maven Central and bintray config

### PR DESCRIPTION
**Pull request by janvanmansum**
### Please ensure you have completed the following before submitting:
- [x] Ran all tests to ensure existing functionality wasn't broken
- [x] Ran all quality assurance checks and fixed any new errors or warnings, which include:
* [PMD](https://pmd.github.io/)
* [FindBugs](http://findbugs.sourceforge.net/)
* [Jacoco](http://eclemma.org/jacoco/) code coverage

**Note: you can complete both boxes by running and fixing warnings/errors with** `gradle clean check`
- [x] Code is [self documenting](https://en.wikipedia.org/wiki/Self-documenting_code) or a short comment when self documenting isn't possible

### CODE REVIEW

File: bintray.gradle [hunk 1]
```diff
@@ -13,20 +13,6 @@ if(!project.hasProperty("ossrhUsername")){ //so CI doesn't break
   project.ext.ossrhPassword = "foo"
 }
 
-task javadocJar(type: Jar) {
```
Comment (janvanmansum) [inline bintray.gradle:16]: Moved this to the master build file.

File: bintray.gradle [hunk 2]
```diff
@@ -38,7 +24,7 @@ bintray {
   pkg {
     repo = "maven"
     name = "bagit-java"
-    userOrg = user
+    userOrg = "dans-knaw"
```
Comment (janvanmansum) [inline bintray.gradle:27]: user and userOrg are not the same for our set-up.

File: bintray.gradle [hunk 3]
```diff
@@ -52,11 +38,6 @@ bintray {
     version{
       name = project.version
       vcsTag = 'v' + project.version
-      mavenCentralSync{
```
Comment (janvanmansum) [inline bintray.gradle:55]: Added commands for automatic publishing, so this doesn't seem necessary anymore.

File: build.gradle [hunk 4]
```diff
@@ -9,8 +9,24 @@ plugins {
   id "org.ajoberstar.grgit" version "2.2.1"
   id "com.github.spotbugs" version "1.6.2"
   id "com.jfrog.bintray" version "1.8.2"
+  id 'io.codearte.nexus-staging' version '0.11.0'
```
Comment (janvanmansum) [inline build.gradle:12]: Plugin for staging to Maven Central.

File: build.gradle [hunk 5]
```diff
@@ -9,8 +9,24 @@ plugins {
...
+}
+
+task sourcesJar(type: Jar) {
+  group "Build"
+  description "Create the jar that contains all the .class files."
+  classifier = 'sources'
+  from sourceSets.main.allSource
+}
+
+//apply from: 'eclipse.gradle'
```
Comment (janvanmansum) [inline build.gradle]: Not sure if we want to keep supporting Eclipse?

File: build.gradle [hunk 6]
```diff
@@ -9,8 +9,24 @@ plugins {
   id "org.ajoberstar.grgit" version "2.2.1"
   id "com.github.spotbugs" version "1.6.2"
   id "com.jfrog.bintray" version "1.8.2"
+  id 'io.codearte.nexus-staging' version '0.11.0'
 }
-apply from: 'eclipse.gradle'
+
```
Comment (janvanmansum) [inline build.gradle:14]: Moved to master build.

File: maven-central.gradle [hunk 7]
```diff
@@ -4,21 +4,6 @@ apply plugin: 'signing'
 
 group = 'nl.knaw.dans'
 
-//javadocs and sources required for uploading to maven central
```
Comment (janvanmansum) [inline maven-central.gradle:7]: Moved to master build.

File: maven-central.gradle [hunk 8]
```diff
@@ -82,3 +67,8 @@ uploadArchives {
     }
   }
 }
+
```
Comment (janvanmansum) [inline maven-central.gradle:70]: For closing and publishing repo from the command line. Kept the comment to document why the staging Profile ID is not hardcoded here.

File: build.gradle (hunk 5 above)
Comment (rvanheest) [reply build.gradle]: Lets keep it in for now. Doesn't hurt anybody, right?

**Pull request merged**